geps/gep-713/index.md

```diff
@@ -1,1157 +1,1094 @@
+
+To avoid any ambiguity in the interpretation of the targets, metaresources MUST clearly define the extent of their effects respectively to the object kinds they target (semantics of attaching a metaresource). This is usually defined in terms of a known hierarchy of resource kinds.
+
+See also: [Declared targets versus Effective targets](#declared-targets-versus-effective-targets) and [Hierarchy of target kinds and Effective metaresources](#hierarchy-of-target-kinds-and-effective-metaresources).
+
+#### Narrowing the target to sections of an object
+
+Metaresource CRDs can offer the option to target a section of an object whose spec defines sections uniquely identifiable by name. These metaresources typically include a field `spec.targetRefs.sectionName` that can be used along with compatible kinds.
+
+E.g. – a metaresource that specifies additional behaviour for a given listener of a Gateway API Gateway object, though not for all listeners of the Gateway, MUST (i) require the Gateway listener to be uniquely named and (ii) provide the `sectionName` field of target reference with the name of the targeted listener.
+
+<details>
+  <summary>Implementation tip</summary>
+
+  This targeting method can be implemented in Golang by using a type such as Gateway API's [`LocalPolicyTargetReferenceWithSectionName`](https://pkg.go.dev/sigs.k8s.io/gateway-api/apis/v1alpha2#LocalPolicyTargetReferenceWithSectionName) type. E.g.:
+
+  ```go
+  package color
+
+  import (
+    metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+    gatewayapiv1alpha2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+  )
+
+  type ColorPolicy struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+
+    // Spec defines the desired state of the policy.
+    Spec ColorPolicySpec `json:"spec"`
+
+    // Status defines the current state of the policy.
+    Status ColorPolicyStatus `json:"status,omitempty"`
+  }
+
+  type ColorPolicySpec struct {
+    // TargetRefs specify the targets of the policy by name.
+    // The following kinds are supported: …
+    // +listType=map
+    // +listMapKey=group
+    // +listMapKey=kind
+    // +listMapKey=name
+    // +listMapKey=sectionName
+    // +kubebuilder:validation:MinItems=1
+    // +kubebuilder:validation:MaxItems=16
+    TargetRefs []gatewayapiv1alpha2.LocalPolicyTargetReferenceWithSectionName `json:"targetRefs"`
+
+    // rest of the spec ("spec proper")…
+  }
+  ```
+</details>
+
+
+#### Targeting virtual types
+
+In some cases, one may want to apply metaresources to objects that are not actual Kubernetes kinds. An example of such, from Gateway API mesh case, would be a hypothetical need for defining a policy to "color" requests to external services. To accomplish this, implementations MAY choose to support a reference to a _virtual resource type_. E.g.:
```

Comment:
Can you start with a definition of "virtual types".  E.g. virtual types are defined as those with a group outside of the typical Kubernetes groups.  As an alternative, I suppose these types could be called custom types.

Reply:
Done